File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s185737640_commented.cpp
#include <iostream>
#include <string>

using namespace std;

bool isEvenString(const string& s) {
    size_t len = s.length();
    if (len % 2 != 0) return false;
    size_t half = len / 2;
    string a = s.substr(0, half);
    string b = s.substr(half);
    return a == b;
}

int main() {
    string S;
    getline(cin, S);

    size_t len = S.length();
    if (len % 2 == 1) {
        S = S.substr(0, len - 1);
    } else {
        S = S.substr(0, len - 2);
    }

    for( ; !S.empty(); ) {
        if (isEvenString(S)) {
            break;
        }
        S = S.substr(0, S.length() - 2);
    }

    cout << S.length() << endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s185737640_commented.py
def is_even_string(s):
    half = len(s) // 2
    return s[:half] == s[half:]

s = input().strip()

if len(s) % 2 == 1:
    s = s[:-1]
else:
    s = s[:-2]

current_s = s

while len(current_s) > 0:
    if is_even_string(current_s):
        break
    else:
        current_s = current_s[:-2]

print(len(current_s))
# End of Code
